### DISCUSSION TITLE
Programming Advices

### DISCUSSION THREAD

**Discussion by LoriaLawrenceZ** [LoriaLawrenceZ/LoriaLawrenceZ#2] [General]
<p>How to Deploy an Angular App to GitHub Pages (like a PRO) - (<a href="https://www.youtube.com/watch?v=F3HbnbT1Maw">Video</a>)</p>
<ul>
    <li><span>Easy way to deploy Angular App/Project in GitHub using GitHub Pages. <a href="https://github.com/LoriaLawrenceZ/LoriaLawrenceZ/discussions/2#discussioncomment-8413153">More info</a></span></li>
</ul>

**Comment by LoriaLawrenceZ** [comment]
<h1>Brief Explanation: Deploying Angular Application with GitHub Pages</h1>

<h2>Command:</h2>
<pre><code>ng build --output-path docs --base-href /ng-deploy</code></pre>

<h2>Purpose:</h2>

<p>This Angular CLI command is like giving instructions to a builder to construct a house. Let's break it down in simpler terms:</p>

<ul>
    <li><code>ng build</code>: This tells Angular CLI to build your Angular application, transforming your code into a format that can be understood by web browsers.</li>
    <li><code>--output-path docs</code>: Here, we're telling Angular CLI where to put the built files. In this example, we're saying "put them in a folder called 'docs'". You'll change 'docs' to wherever you want to put your application when you deploy it.</li>
    <li><code>--base-href /ng-deploy</code>: This sets the base URL for your application. Imagine your application is a book, and each page is a different part of your app. This command is saying "start reading from the page called '/ng-deploy'". You'll replace '/ng-deploy' with whatever URL your application will be served from when it's deployed.</li>
</ul>

<h2>Summary:</h2>

<p>So, when you run this command, Angular CLI will build your application and put the files in a folder (like packaging up your house into boxes). It'll also make sure the application knows where it's going to live on the internet. This command helps prepare your Angular application for deployment to a web server or hosting service.</p>

**Comment by LoriaLawrenceZVR** [comment]
Money Printer (<a href="https://github.com/FujiwaraChoki/MoneyPrinter">Repository</a>)
